tumblr/XExtensionItem
# Add a framework target (and as such, Carthage support)

https://github.com/tumblr/XExtensionItem/issues/42

Diverge from the traditional `pod lib create` project structure in order to add first class framework support.

* `XExtensionItem.xcodeproj` in the root directory now has a framework target and the tests target.
* `Example/XExtensionItemExample.xcworkspace` contains a sample application and share extension, and pulls XExtensionItem in as a development pod

Also fixes a bug in `XExtensionItem.m` and updates the unit tests such that they will fail if it returns.

## To test

* Verify that the framework target builds without any warnings or errors, and that tests all pass
* Verify that the sample application and share extension function as expected
* Verify that a sample project is able to consume XExtensionItem via [Carthage](https://github.com/carthage/carthage)
    * Clone the [CarthageTest](https://github.com/irace/CarthageTest) project locally (it won’t build until the following steps are completed)
    * `brew install carthage`
    * From the `CarthageTest` directory, run `carthage update`
    * Follow [these instructions from the Carthage README](https://github.com/carthage/carthage#if-youre-building-for-ios)
    * Verify that CarthageTest builds properly
